test/crypto-service/test_accountCreateTransaction.js

```diff
@@ -0,0 +1,1005 @@
+      verifyAccountCreationWithMaxAutoTokenAssociations(response.accountId, maxAutoTokenAssociations)
+    });
+
+    it("(#2) Creates an account with a max token association set to 0", async function () {
+      // Generate a valid key for the account.
+      const key = await JSONRPCRequest("generateKey", {
+        type: "ed25519PublicKey"
+      });
+      if (key.status === "NOT_IMPLEMENTED") this.skip();
+      
+      // Attempt to create an account with the max automatic token associations set to 0.
+      const maxAutoTokenAssociations = 0;
+      const response = await JSONRPCRequest("createAccount", {
+        key: key.key,
+        maxAutoTokenAssociations: maxAutoTokenAssociations,
+      });
+      if (response.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Verify the account was created with the max automatic token associations set to 0.
+      verifyAccountCreationWithMaxAutoTokenAssociations(response.accountId, maxAutoTokenAssociations)
+    });
+
+    it("(#3) Creates an account with a max token association that is the maximum value", async function () {
+      // Generate a valid key for the account.
+      const key = await JSONRPCRequest("generateKey", {
+        type: "ecdsaSecp256k1PrivateKey"
+      });
+      if (key.status === "NOT_IMPLEMENTED") this.skip();
+      
+      // Attempt to create an account with the max automatic token associations set to the maximum value.
+      const maxAutoTokenAssociations = 5000;
+      const response = await JSONRPCRequest("createAccount", {
+        key: key.key,
+        maxAutoTokenAssociations: maxAutoTokenAssociations,
+        commonTransactionParams: {
+          maxTransactionFee: 100000000000
+        }
+      });
+      if (response.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Verify the account was created with the max automatic token associations set to 5000.
+      verifyAccountCreationWithMaxAutoTokenAssociations(response.accountId, maxAutoTokenAssociations)
+    });
+
+    it("(#4) Creates an account with a max token association that is the maximum value plus one", async function () {
+      // Generate a valid key for the account.
+      const key = await JSONRPCRequest("generateKey", {
+        type: "ed25519PrivateKey"
+      });
+      if (key.status === "NOT_IMPLEMENTED") this.skip();
+
+      try {
+        // Attempt to create an account with the max automatic token associations over the maximum value. The network should respond with an REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT status.
+        const response = await JSONRPCRequest("createAccount", {
+          key: key.key,
+          maxAutoTokenAssociations: 5001,
+          commonTransactionParams: {
+            maxTransactionFee: 100000000000
+          }
+        });
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch (err) {
+        assert.equal(err.data.status, "REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT");

```

Comment:
I think this status code was changed in https://github.com/hashgraph/hedera-services/issues/13687 . Now it returns `INVALID_MAX_AUTO_ASSOCIATIONS`

Reply:
Yep, I can see this test failing now when testing with `--network-tag=0.52.0-alpha.5`. I'll fix this here when I can test it against C++ server

Reply:
Updated to new response code and working with C++ server

Reply:
Why did this get reverted?

Reply:
This was a mistake, reverted back